flyteadmin/pkg/async/webhook/implementations/slack_webhook.go

```diff
@@ -0,0 +1,78 @@
+package implementations
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/flyteorg/flyte/flyteidl/gen/pb-go/flyteidl/admin"
+	"github.com/flyteorg/flyte/flytepropeller/pkg/controller/nodes/task/secretmanager"
+
+	"github.com/flyteorg/flyte/flytestdlib/logger"
+
+	"github.com/flyteorg/flyte/flyteadmin/pkg/async/webhook/interfaces"
+	runtimeInterfaces "github.com/flyteorg/flyte/flyteadmin/pkg/runtime/interfaces"
+	"github.com/flyteorg/flyte/flytestdlib/promutils"
+)
+
+const Slack = "slack"
+
+type SlackWebhook struct {
+	Config        runtimeInterfaces.WebHookConfig
+	systemMetrics webhookMetrics
+}
+
+func (s *SlackWebhook) GetConfig() runtimeInterfaces.WebHookConfig {
+	return s.Config
+}
+
+func (s *SlackWebhook) Post(ctx context.Context, payload admin.WebhookMessage) error {
```

Comment:
Need to rename it to webhookClient or something else? it should work if people give any other kind of webhook endpoints.